service/k8sclient/k8client_test.go

```diff
@@ -36,75 +37,174 @@ func TestK8Client(t *testing.T) {
 	validKubeconfig, err := kubeCtl.Run(ctx, []string{"config", "view", "-o", "json"}, nil)
 	require.NoError(t, err)
 
-	client, err := NewK8Client(ctx, string(validKubeconfig))
+	client, err := New(ctx, string(validKubeconfig))
 	require.NoError(t, err)
 	t.Cleanup(func() {
 		err := client.Cleanup()
 		require.NoError(t, err)
 	})
 
-	name := "test-cluster"
-	_ = client.DeleteXtraDBCluster(ctx, name)
+	l := logger.Get(ctx)
 
-	for {
-		clusters, err := client.ListXtraDBClusters(ctx)
+	t.Run("XtraDB", func(t *testing.T) {
+		name := "test-cluster"
+		_ = client.DeleteXtraDBCluster(ctx, name)
```

Comment:
This takes some time - may be better sleep right here but not at the end of the loop, or before calling client.ListXtraDBClusters?

Reply:
Yeah, it makes sense